lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php

```diff
@@ -1619,4 +1619,18 @@ private function generateIdentifierName($identifier)
 
         return strtoupper(dechex(crc32($identifier->getName())));
     }
+
+    protected function getDropPrimaryKeySQL(string $table) : string
+    {
+        return "
+DECLARE @table NVARCHAR(512), @sql NVARCHAR(MAX);
+SELECT @table = N'{$table}';
+SELECT @sql = 'ALTER TABLE ' + @table + ' DROP CONSTRAINT '  + name + ';'
+    FROM sys.key_constraints
+    WHERE [type] = 'PK'
+    AND [parent_object_id] = OBJECT_ID(@table);
```

Comment:
Would it make sense to move the object ID outside of the `@sql` so that the function is not invoked for every single PK in `sys.key_constraints`?

How will this query behave if the table doesn't have a PK? I'd expect `sp_executeSQL` be invoked with an empty argument and throwing some sort of exception.

Reply:
Emm. If you will try to drop PK for any other platforms on table not containing it, exception will be thrown as well, so I don't see any inconsistency here.

Reply:
Is it the same type of exception?

Reply:
Hmm, different one, you are right. Just `syntax error` vs `can't drop unexisted PK`. What's your proposal? I don't have windows pc and that is my very first mssql experience in life :)

Reply:
You don't need a Windows PC. The SQL Server extension can be [compiled for PHP on Linux](https://github.com/Microsoft/msphpsql/releases/tag/v5.2.0-RC), and the server itself can be running in a Docker container ([microsoft/mssql-server-linux](https://hub.docker.com/r/microsoft/mssql-server-linux/)).

I don't have a proposal currently. My only concern is this code may execute knowingly syntactically invalid SQL. Instead of doing the in the dynamic SQL way, can we just rely on some conventional PK name in SQL Server (like `DROP CONSTRAINT PK_$table;` from [here](https://docs.microsoft.com/en-us/sql/relational-databases/tables/delete-primary-keys#TsqlProcedure))?

Reply:
Hold on. `getDropPrimaryKeySQL()` is not even part of the abstract platform API. Why are we using it in the abstract platform code? From what I see, most of the platforms are capable of identifying PK by name including SQL Server. Why should we take special care about the PK on SQL Server if it can be just dropped by name same as other constraints?

Reply:
We can't  rely on any name conventions/best practices. As far as I investigated, the name builds according to schema `PK_{$tableName}_{$someHash}` and can't be predicted.